### PR TITLE
Add documentation for 'nstack test' command

### PR DESCRIPTION
**Pull request by NickPollard**

### CODE REVIEW

File: reference/nstack_toolkit.rst [hunk 1]
```diff
@@ -178,7 +178,44 @@ Used with the HTTP source, ``nstack send`` sends a JSON-encoded element to an en
...
+^^^^^^^^
+
+.. code:: bash
+
+    $ nstack test <function> <input>
+
+=============    ===========
+Option           Description
+=============    ===========
+``function``     A fully-qualified function name, e.g. ``Foo:0.0.1.numChars``
```
Comment (UnkindPartition) [inline reference/nstack_toolkit.rst]: The syntax doesn't look right, module and function names are given separately.

```
% nstack test --help                                                                                                                                                                         
Usage: nstack test module function_name event
  Test executing function with a single given json-snippet value

Available options:
  module                   Module name
  function_name            Function name
  event                    JSON Snippet to send as an event
  -h,--help                Show this help text
```

File: reference/nstack_toolkit.rst [hunk 2]
```diff
@@ -178,7 +178,44 @@ Used with the HTTP source, ``nstack send`` sends a JSON-encoded element to an en
...
+The test data should be JSON-encoded,
+just as with `nstack send`. 
+The test data must be of the type the function expects
+or it will be rejected by the nstack server.
+
+.. code:: bash
+
+    $ nstack test Foo:0.0.1.numChars '"Hello World"'
 
+    $ nstack test Foo:0.0.1.sum '[1,2,3]'
```
Comment (UnkindPartition) [inline reference/nstack_toolkit.rst]: These examples' syntax needs fixing, too.

File: reference/nstack_toolkit.rst [hunk 3]
```diff
@@ -178,7 +178,44 @@ Used with the HTTP source, ``nstack send`` sends a JSON-encoded element to an en
...
+The function will be started,
+called with the test data to process,
+and then stopped.
+
+Test is equivalent to starting the function with an HTTP source
+and log sink,
+using `nstack send` to send an input value,
+and then stopping the process.
+
+It is very useful for unit-testing a function
```
Comment (lanthias) [inline reference/nstack_toolkit.rst]: I'd say 'Can be used'

File: reference/nstack_toolkit.rst (hunk 1 above)
Comment (NickPollard) [reply reference/nstack_toolkit.rst]: Yeah I realised this after I submitted this, I forgot that I'd changed the syntax (originally they were specified together) of the command and I wrote this for the old one. Whoops!

**Pull request merged**